mapper.go

```diff
@@ -733,6 +735,30 @@ func fileContentMapper(r *Registry) MapperFunc {
 	}
 }
 
+func regexMapper() MapperFunc {
```

Comment:
`regexp.Regexp` actually implements `encoding.Text(Un)marshaler`, so I don't think this is necessary either after all...

Reply:
oh boy, you're right! :facepalm: 
totally overlooked that too :see_no_evil:

Reply:
sorry for consuming you time 🤦

i removed all mappers. we can close those this pr.

I added some regex tests in the last commit, if you have some time, maybe you can explain to me why the TestRegex,TestRegexSlice,TestRegexPointer are working but the TestRegexPointerSlice is not?
- somehow i used to always use `*regex.Regexp` instead of `regex.Regexp` but the pointer version does not with the slice